wolf_engine_core/src/events/engine_events.rs

```diff
@@ -6,7 +6,7 @@ pub enum WindowEvent {}
 /// Provides the main events used by Wolf Engine.
 #[non_exhaustive]
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
```

Comment:
Maybe try removing some of these extra types?

```suggestion
#[derive(Debug, PartialEq)]
```